### ISSUE TITLE
TensorflowNer: Evaluate test set after all epochs complete

### ISSUE THREAD

**Issue by vosmith** [JohnSnowLabs/spark-nlp#2321]
**Is your feature request related to a problem? Please describe.**
It seems that both the validation and test sets are run at the end of each epoch.  From what I've seen in common practice is the validation set should be run at the end of each epoch, but the test set should only be evaluated once all epochs have completed.  I'm not going to say that this is an absolute must, however it struck me as a bit odd and I think it is worth a change.

**Describe the solution you'd like**
Move test set evaluation to after all epochs are complete

**Describe alternatives you've considered**
Leaving the code as is.

**Additional context**
I am willing to work on a PR if there is a general consensus

**Comment by maziyarpanahi** [comment]
Hi,
It’s not really a rule nor changes anything in the last outcome. The evaluation is also very fast.
If the users really want to have the test dataset at the very end for any given reason, they can just use any library at the end to get the classification metrics using the test dataset. (We have many examples for this where we didn’t provide the test dataset and used sklearn at the end)

However, the choice of having test dataset metrics during the training is for a reason. Validation is based on a proportion of training dataset, if the training dataset is small, unbalanced, etc. this will bias the metrics during the training. You would think after 2 hours the model performed perfectly only to find out at the very end that it didn’t! Having an option to have an entirely new dataset on each Epoch gives valuable information regarding overfitting/under-fitting. You can stop and tune the parameters or even the dataset.

Thanks for offering help, but we have decided to design it this way out of experience and feedbacks we received.
